Region/BasicStatsCalculator.tcc

```diff
@@ -0,0 +1,100 @@
+#ifndef CARTA_BACKEND_REGION_BASICSTATSCALCULATOR_TCC_
+#define CARTA_BACKEND_REGION_BASICSTATSCALCULATOR_TCC_
+
+#include "BasicStatsCalculator.h"
+
+namespace carta {
+
+template<typename T>
+void BasicStats<T>::join(BasicStats<T>& other) {
+    if (other.num_pixels) {
+        sum += other.sum;
+        sumSq += other.sumSq;
+        num_pixels += other.num_pixels;
+        min_val = std::min(min_val, other.min_val);
+        max_val = std::max(max_val, other.max_val);
+        mean = sum / num_pixels;
+        stdDev = num_pixels > 1 ? sqrt((sumSq - (sum * sum / num_pixels)) / (num_pixels - 1)): NAN;
```

Comment:
Casacore returns zero if the number of pixels is <= 1.5; not sure if we absolutely have to match that behaviour.

Reply:
that's a good question. What's the standard deviation of a sample of one?

Reply:
if the _population_ is 1, then the standard deviation should be zero, that makes sense. But casacore is using the _sample_ standard deviation (I'm not 100% sure why, or whether we should be using the sample std dev for regions, and population std dev for entire images), in which case the standard deviation of a sample would be undefined.

Reply:
paging @kswang1029 for information on when it makes sense to use sample versus population when it comes to regions containing a subset of the image, and regions containing the entire image.

Reply:
Not sure where the 1.5 comes from. `n` is a double, but (surely) is an integer value -- maybe this is avoiding a floating point precision error.